src/ghost_cursor.rs

```diff
@@ -182,6 +182,7 @@ impl<'a, 'brand, T> GhostCursor<'a, 'brand, T> {
     ///     }
     /// });
     /// ```
+    #[allow(clippy::result_unit_err)]
     pub fn move_mut<F>(&mut self, fun: F) -> Result<(), ()>
```

Comment:
I allowed this so that the `lint  job passes, perhaps the return type should be changed instead.

Reply:
I agree that the return type should be modified, but this is off-topic for this PR so I think you took the correct step in allowing the lint for now.